scripts/popup.js

```diff
@@ -1,3 +1,98 @@
+function getFormData(){
+    var formElements = document.getElementById('metadata-form').elements;
+    var formData = {};
+
+    for(var i = 0 ; i < formElements.length ; i++){
+        var input = formElements[i];
+        if( input.type != "button" ){
+            formData[ input.name ] = input.value;
+        }
+    }
+
+    return formData;
+}
+
+function postDataToApi( xhr, formData ){
+    xhr.open("POST", "http://localhost:3000/public/documents", true);
+    xhr.setRequestHeader('Content-Type', 'application/json');
+
+    var data = JSON.stringify(formData)
+
+    xhr.send(data);
+}
+
+function click(e) {
+    var formData = getFormData();
+    var xhr = new XMLHttpRequest();
+
+    postDataToApi(xhr, formData);
+
+    xhr.onreadystatechange = function () {
+        if (this.readyState != 4) return;
+
+        if (this.status == 200) {
+            var data = JSON.parse(this.responseText);
+
+            // we get the returned data
+        }
+
+        // end of state change: it can be after some time (async)
+    };
+}
+
+function getMetadataFields(){
+    return ([
+                { name: "Author", type: "text" },
+                { name: "Title", type: "text" }
+           ]);
+}
+
+function generateMetadataInput(metadataField){
+    var inputTemplate = document.querySelector('#input-template');
+    var label = inputTemplate.content.querySelector('label');
+    var input = inputTemplate.content.querySelector('input');
```

Comment:
Instead of using `<label></label><input>` as a template and copying off of it, you can do something like:
```
var $label = $('<label></label>');
var $input = $('<input>');
```
You can also put info into the tags like:
```
var $label = $('<label>' + metadataField["name"] + '</label>');
```